### PR TITLE
Treemap + Sunburst

### PR DESCRIPTION
**Pull request by du-phan**

### CODE REVIEW

File: plugin.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+    "id": "dss-plugin-hierarchical-charts",
+    "version": "0.1.0",
+    "meta": {
+        "label": "Hierarchical charts",
+        "description": "",
```
Comment (JoachimZ) [inline plugin.json:6]: "Treemap and Sunburst plugin charts"

File: plugin.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+    "id": "dss-plugin-hierarchical-charts",
```
Comment (JoachimZ) [inline plugin.json]: hierarchical-charts

File: plugin.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    "version": "0.1.0",
+    "meta": {
+        "label": "Hierarchical charts",
+        "description": "",
+        "author": "Dataiku (Du Phan)",
+        "icon": "icon-bar-chart",
+        "licenseInfo": "Apache Software License",
+        "url": "https://www.dataiku.com/dss/plugins/info/google-charts.html",
```
Comment (JoachimZ) [inline plugin.json]: https://www.dataiku.com/dss/plugins/info/hierarchical-charts.html

File: webapps/sunburst-chart/app.js [hunk 4]
```diff
@@ -0,0 +1,276 @@
+// Dimensions of sunburst.
+var width = 350;
```
Comment (JoachimZ) [inline webapps/sunburst-chart/app.js]: tiny detail, I would avoid ever using "var" use "let" or "const" instead, it avoids a few bugs sometimes. It's really not much but it does not cost anything...

File: webapps/sunburst-chart/app.js [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+          initializeBreadcrumbTrail();
+    }  
+    
+    // Bounding circle underneath the sunburst, to make it easier to detect
+    // when the mouse leaves the parent g.
+    vis.append("svg:circle")
+        .attr("r", radius)
+        .style("opacity", 0);
+
+    // For efficiency, filter nodes to keep only those large enough to see.
```
Comment (JoachimZ) [inline webapps/sunburst-chart/app.js:51]: Nice! (0.29 degree seems small but I did not check yet)

File: webapps/sunburst-chart/app.js [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+    
+    var uniqueNames = (function(a) {
+        var output = [];
+        a.forEach(function(d) {
+            if (output.indexOf(d.name) === -1) {
+                output.push(d.name);
+            }
+        });
+        return output;
+    })(nodes);
```
Comment (JoachimZ) [inline webapps/sunburst-chart/app.js]: Mwarf, it seems correct but circumvoluted :D 

`const uniqueNames = Array.from(new Set(nodes.map(n => n.name)))`
or 
`const uniqueNames = [...new Set(nodes.map(n => n.name))]`
or 
```
function unique(arr) {
    return [...new Set(arr)];
}
const uniqueNames = unique(nodes.map(n => n.name));
```

File: webapps/sunburst-chart/webapp.json [hunk 7]
```diff
@@ -0,0 +1,46 @@
+{
+    "meta": {
+        "label": "Sunburst chart",
+        "description": "",
```
Comment (JoachimZ) [inline webapps/sunburst-chart/webapp.json]: Visualize hierarchies as concentric circles

File: webapps/sunburst-tree/app.js [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+    
+    var uniqueNames = (function(a) {
+        var output = [];
+        a.forEach(function(d) {
+            if (output.indexOf(d.name) === -1) {
+                output.push(d.name);
+            }
+        });
+        return output;
+    })(nodes);
```
Comment (JoachimZ) [inline webapps/sunburst-tree/app.js]: same as above

File: webapps/sunburst-chart/webapp.json [hunk 9]
```diff
@@ -0,0 +1,46 @@
+{
+    "meta": {
+        "label": "Sunburst chart",
```
Comment (JoachimZ) [inline webapps/sunburst-chart/webapp.json]: Just "Sunburst"

File: webapps/sunburst-tree/webapp.json [hunk 10]
```diff
@@ -0,0 +1,20 @@
+{
+    "meta": {
+        "label": "Sunburst tree",
```
Comment (JoachimZ) [inline webapps/sunburst-tree/webapp.json]: Just "Sunburst"

File: webapps/treemap-chart/body.html [hunk 11]
```diff
@@ -0,0 +1,9 @@
+<script src="/plugins/dss-plugin-hierarchical-charts/resource/dku-helpers.js"></script>
+<script src="/plugins/dss-plugin-hierarchical-charts/resource/treemap-chart.js"></script>
```
Comment (JoachimZ) [inline webapps/treemap-chart/body.html]: does not exist, right?

File: webapps/treemap-chart/style.css [hunk 12]
```diff
@@ -0,0 +1 @@
+@import '/plugins/dss-plugin-hierarchical-charts/resource/style.css';
```
Comment (JoachimZ) [inline webapps/treemap-chart/style.css]: update without dss-plugin

File: Makefile [hunk 13]
```diff
@@ -0,0 +1,8 @@
+PLUGIN_VERSION=0.1.0
+PLUGIN_ID=dss-plugin-hierarchical-charts
```
Comment (JoachimZ) [inline Makefile]: update without dss-plugin

File: webapps/sunburst-chart/body.html [hunk 14]
```diff
@@ -0,0 +1,12 @@
+<script src="/plugins/dss-plugin-hierarchical-charts/resource/dku-helpers.js"></script>
```
Comment (JoachimZ) [inline webapps/sunburst-chart/body.html]: update without dss-plugin

File: webapps/sunburst-chart/style.css [hunk 15]
```diff
@@ -0,0 +1,46 @@
+@import '/plugins/dss-plugin-hierarchical-charts/resource/style.css';
```
Comment (JoachimZ) [inline webapps/sunburst-chart/style.css:1]: update without dss-plugin

File: webapps/sunburst-tree/body.html [hunk 16]
```diff
@@ -0,0 +1,12 @@
+<script src="/plugins/dss-plugin-hierarchical-charts/resource/dku-helpers.js"></script>
```
Comment (JoachimZ) [inline webapps/sunburst-tree/body.html]: update without dss-plugin

File: webapps/sunburst-tree/style.css [hunk 17]
```diff
@@ -0,0 +1,46 @@
+@import '/plugins/dss-plugin-hierarchical-charts/resource/style.css';
```
Comment (JoachimZ) [inline webapps/sunburst-tree/style.css]: update without dss-plugin

File: webapps/treemap-chart/body.html [hunk 18]
```diff
@@ -0,0 +1,9 @@
+<script src="/plugins/dss-plugin-hierarchical-charts/resource/dku-helpers.js"></script>
```
Comment (JoachimZ) [inline webapps/treemap-chart/body.html]: update without dss-plugin

**Review by JoachimZ — CHANGES_REQUESTED** [review]
Why columns on the left, not on top? To spare vertical space?
Error reporting does not seem correct, make sure the user has feedback when webapp backend fails (for any reason)
I can probably make a few more comment.

**Comment by du-phan** [comment]
I made all the modifications according to Joachim's review

**Comment by du-phan** [comment]
> Why columns on the left, not on top? To spare vertical space?

Yes so that we have places for the rules dropdown

**Comment by ghost** [comment]
Tested and reviewed, very nice 👍

**Pull request merged**